File: BussinessLogic/LanguagesBO.cs
using System;
using System.Collections.Generic;
using System.Data.Entity.Migrations;
using System.Linq;
using DataAccess;

// ReSharper disable once CheckNamespace

namespace BussinessLogic {
    /// <summary>
    /// LanguagesBO
    /// </summary>
    /// <author> Nam Chu </author>
    /// <created>10/06/15</created>
    /// <modified>
    /// Nam Chu
    /// 01/07/15
    /// 
    /// -reformat and optimize code
    /// </modified>
    
    // ReSharper disable once InconsistentNaming
    public class LanguagesBO {
        private readonly DatabaseDA _aDatabaseDa = new DatabaseDA();

        #region SELECT

        public Languages Sel_ByID(Int32 id) {
            try {
                Languages aLanguages = _aDatabaseDa.Languages.Where(b => b.ID == id).ToList()[0];
                return aLanguages;
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel: {0}", ex.Message));
            }
        }

        public Languages Sel_ByID(Int32 id, bool disable) {
            try {
                Languages aLanguages =
                    _aDatabaseDa.Languages.Where(b => b.ID == id).Where(b => b.Disable == disable).ToList()[0];
                return aLanguages;
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel: {0}", ex.Message));
            }
        }

        public List<Languages> Sel() {
            try {
                return _aDatabaseDa.Languages.ToList();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel:{0}", ex.Message));
            }
        }

        public List<Languages> Sel(bool disable) {
            try {
                return _aDatabaseDa.Languages.Where(z => z.Disable == disable).ToList();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel:{0}", ex.Message));
            }
        }

        public List<Languages> Sel_ByType(int type) {
            try {
                return _aDatabaseDa.Languages.Where(p => p.Type == type).ToList();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel_ByType:{0}", ex.Message));
            }
        }

        public List<Languages> Sel_ByType(int type, bool disable) {
            try {
                return _aDatabaseDa.Languages.Where(p => p.Type == type).Where(z => z.Disable == disable).ToList();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel_ByType:{0}", ex.Message));
            }
        }

        public List<Languages> Sel_ByStatus(int status) {
            try {
                return _aDatabaseDa.Languages.Where(p => p.Status == status).ToList();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel_ByStatus:{0}", ex.Message));
            }
        }

        public List<Languages> Sel_ByStatus(int status, bool disable) {
            try {
                return _aDatabaseDa.Languages.Where(p => p.Status == status).Where(z => z.Disable == disable).ToList();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel_ByStatus:{0}", ex.Message));
            }
        }

        public List<Languages> Sel_ByType_ByStatus(int type, int status) {
            try {
                return _aDatabaseDa.Languages.Where(c => c.Type == type).Where(p => p.Status == status).ToList();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel_ByType_ByStatus:{0}", ex.Message));
            }
        }

        public List<Languages> Sel_ByType_ByStatus(int type, int status, bool disable) {
            try {
                return
                    _aDatabaseDa.Languages.Where(c => c.Type == type)
                        .Where(p => p.Status == status)
                        .Where(z => z.Disable == disable)
                        .ToList();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Sel_ByType_ByStatus:{0}", ex.Message));
            }
        }

        #endregion

        #region DELETE

        public int Del_ByID(Int32 id) {
            try {
                List<Languages> aTemp = _aDatabaseDa.Languages.Where(p => p.ID == id).ToList();
                if (aTemp.Count > 0) {
                    _aDatabaseDa.Languages.Remove(aTemp[0]);
                    return _aDatabaseDa.SaveChanges();
                }
                throw new Exception(String.Format("LanguagesBO.Del_ByID: {0}", "Không tìm thấy Languages có ID = " + id));
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Del: {0}", ex.Message));
            }
        }

        public int Del_ByListID(List<int> aListId) {
            try {
                List<Languages> aTemp =
                    _aDatabaseDa.Languages.Where(p => aListId.Contains(int.Parse(p.ID.ToString()))).ToList();
                if (aTemp.Count > 0) {
                    _aDatabaseDa.Languages.RemoveRange(aTemp);
                    return _aDatabaseDa.SaveChanges();
                }
                return 0;
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Del_By_ListID: {0}", ex.Message));
            }
        }

        #endregion

        #region UPDATE

        public int Upd_Status(int id, int newStatus) {
            try {
                List<Languages> aListLanguages = _aDatabaseDa.Languages.Where(b => b.ID == id).ToList();
                if (aListLanguages.Count > 0) {
                    aListLanguages[0].Status = newStatus;
                    _aDatabaseDa.Languages.AddOrUpdate(aListLanguages[0]);
                    return _aDatabaseDa.SaveChanges();
                }
                return 0;
            }
            catch (Exception ex) {
                throw new Exception(String.Format("Upd_Status_ByID:{0}", ex.Message));
            }
        }

        public int Upd_Type(int id, int newType) {
            try {
                List<Languages> aListLanguages = _aDatabaseDa.Languages.Where(b => b.ID == id).ToList();
                if (aListLanguages.Count > 0) {
                    aListLanguages[0].Type = newType;
                    _aDatabaseDa.Languages.AddOrUpdate(aListLanguages[0]);
                    return _aDatabaseDa.SaveChanges();
                }
                return 0;
            }
            catch (Exception ex) {
                throw new Exception(String.Format("Upd_Type_ByID:{0}", ex.Message));
            }
        }

        public int Upd_Disable(int id, bool newDisable) {
            try {
                List<Languages> aListLanguages = _aDatabaseDa.Languages.Where(b => b.ID == id).ToList();
                if (aListLanguages.Count > 0) {
                    aListLanguages[0].Disable = newDisable;
                    _aDatabaseDa.Languages.AddOrUpdate(aListLanguages[0]);
                    return _aDatabaseDa.SaveChanges();
                }
                return 0;
            }
            catch (Exception ex) {
                throw new Exception(String.Format("Upd_Disable_ByID:{0}", ex.Message));
            }
        }

        public int Upd_ByListID(List<Languages> aListLanguages) {
            try {
                _aDatabaseDa.Languages.AddOrUpdate(aListLanguages.ToArray());
                return _aDatabaseDa.SaveChanges();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Ins: {0}", ex.Message));
            }
        }

        public int Upd(Languages aLanguages) {
            try {
                _aDatabaseDa.Languages.AddOrUpdate(aLanguages);
                return _aDatabaseDa.SaveChanges();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Upd: {0}", ex.Message));
            }
        }

        #endregion

        #region INSERT

        public int Ins(Languages aLanguages) {
            try {
                _aDatabaseDa.Languages.AddOrUpdate(aLanguages);
                return _aDatabaseDa.SaveChanges();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Ins: {0}", ex.Message));
            }
        }

        public int Ins_ListLanguages(ref List<Languages> aListLanguages) {
            try {
                aListLanguages = _aDatabaseDa.Languages.AddRange(aListLanguages).ToList();
                return _aDatabaseDa.SaveChanges();
            }
            catch (Exception ex) {
                throw new Exception(String.Format("LanguagesBO.Ins: {0}", ex.Message));
            }
        }

        #endregion
    }
}

File: CMS/JS/FRAMEWORK/BussinessLogic/LanguagesBO.js
/**
 * @author Nam Chu
 * @created 02/0/15
 * 
 * @modified Nam Chu 08/07/15
 * -add some methods to work with ActionHandler
 * -add disable to Sel methods
 * -reformat and optimize code
 */
//TODO: Upd_Type, Upd_Status, Upd_Disable are NOT available
var LanguagesBO = {
    Sel: function(disable) {
        var ret;
        if (disable == undefined) {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel",
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        } else {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel&Disable=" + disable,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        }
        return ret;
    },

    Sel_ByID: function(id, disable) {
        var ret;
        if (disable == undefined) {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel_ByID&ID=" + id,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        } else {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel_ByID&ID=" + id + "&Disable=" + disable,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        }
        return ret;
    },

    Sel_ByType: function(type, disable) {
        var ret;
        if (disable == undefined) {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel_ByType&Type=" + type,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        } else {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel_ByType&Type=" + type + "&Disable=" + disable,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        }
        return ret;
    },

    Sel_ByStatus: function(status, disable) {
        var ret;
        if (disable == undefined) {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel_ByStatus&Status=" + status,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        } else {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel_ByStatus&Status=" + status + "&Disable=" + disable,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        }
        return ret;
    },

    Sel_ByType_ByStatus: function(type, status, disable) {
        var ret;
        if (disable == undefined) {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel_ByType_ByStatus&Type=" + type + "&Status=" + status,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        } else {
            jQuery.ajax({
                url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Sel_ByType_ByStatus&Type=" + type + "&Status=" + status + "&Disable=" + disable,
                type: "POST",
                dataType: "json",
                data: "",
                success: function(data, dataStatus) {
                    ret = data;
                },
                timeout: 30000,
                async: false,
                error: function(request, error) {}
            });
        }
        return ret;
    },

    Ins: function () {
        EnableLoading();
        jQuery.ajax({
            url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Ins",
            type: "POST",
            dataType: "json",
            data: $("#frmIns_Languages").serialize(),
            success: function(data) {
                if (data.status == "success")
                    alert("Thêm Languages thành công.");
                else if (data.status != "success")
                    alert("Thêm Languages lỗi: <font style='font-size:9px'>" + data.message + "</font>");
                DisableLoading();
            },

            error: function(ex) {}
        });
    },

    Upd_ByID: function () {
        EnableLoading();
        jQuery.ajax({
            url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Upd_ByID",
            type: "POST",
            dataType: "json",
            data: $("#frmUpd_Languages").serialize(),
            success: function(data) {
                if (data.status == "success")
                    alert("Update Languages thành công.");
                else if (data.status != "success")
                    alert("Update Languages lỗi: <font style='font-size:9px'>" + data.message + "</font>");
                DisableLoading();
            },
            error: function(ex) {}
        });
    },

    Upd_Type_ByID: function () {
        alert("Function Not Implemented");
    },

    Upd_Status_ByID: function () {
        alert("Function Not Implemented");
    },

    Upd_Disable_ByID: function () {
        alert("Function Not Implemented");
    },

    Del_ByID: function (id) {
        EnableLoading();
        jQuery.ajax({
            url: "/Action/ProcessBackendAction.ashx?ActionObject=Languages&action=Del_ByID&ID=" + id,
            type: "POST",
            dataType: "json",
            data: $("#frmUpd_Languages").serialize(),
            success: function(data) {
                if (data.status == "success")
                    alert("Delete thành công.");
                else if (data.status != "success")
                    alert("Delete lỗi: <font style='font-size:9px'>" + data.message + "</font>");
                DisableLoading();
            },
            error: function(ex) {}
        });
    }
}